### PR TITLE
global: PostgreSQL Text type compatibility fix 

### LINKED ISSUE #3037: sqlalchemy: PostgreSQL - type Text and Integer compatibility
PostgreSQL Text type doesn't accept length parameter: http://www.postgresql.org/docs/9.1/static/datatype-character.html

```python
ProgrammingError: (psycopg2.ProgrammingError) type modifier is not allowed for type "text"
LINE 5:  value TEXT(35) NOT NULL,
                ^
[SQL: "\nCREATE TABLE bib00x (\n\tid SERIAL NOT NULL, \n\ttag VARCHAR(6) DEFAULT '' NOT NULL, \n\tvalue TEXT(35) NOT NULL, \n\tPRIMARY KEY (id)\n)\n\n"]
```

I tried to ask to zzzeek if exists a easy solution to this problem: https://bitbucket.org/zzzeek/sqlalchemy/issue/3370/text-and-length

In alternative, with @jirikuncar, we agree to resolve in the same way as for [Integer](https://github.com/inveniosoftware/invenio/blob/master/invenio/ext/sqlalchemy/types/legacyinteger.py#L26)

### PR DESCRIPTION
**Pull request by hachreak**
* BETTER Improves compatibility of Text fields  in PostrgeSQL by
  changing Text in models and removes Invenio hacks on MySQL Index and
  Primary Key creation because starting from SQLAlchemy>=1.0 it arises
  an exception if the length is specified. (closes #3037)

* Updates minimum Flask-Admin version to 1.1.0.

* PEP8/257 code style improvements.

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: invenio/modules/access/models.py [hunk 1]
```diff
@@ -235,8 +253,9 @@ class UserAccROLE(db.Model):
     ).scalar()
 )
 
-User.has_super_admin_role = property(lambda self:
-    db.object_session(self).query(db.func.count(User.id)>0).join(
+User.has_super_admin_role = property(
+    lambda self:
```
Comment (jirikuncar) [inline invenio/modules/access/models.py]:
```python
    lambda self: db.object_session(self).query(
        db.func.count(User.id) > 0).join(
```

File: invenio/modules/messages/models.py [hunk 2]
```diff
@@ -85,108 +94,121 @@ def validate_sent_to_user_nicks(self, key, value):
 
     @db.validates('_sent_to_group_names')
     def validate_sent_to_group_names(self, key, value):
-        group_names = filter(len, map(strip,
-            value.split(cfg['CFG_WEBMESSAGE_SEPARATOR'])))
+        """Validate sent to group names."""
+        group_names = filter(
```
Comment (jirikuncar) [inline invenio/modules/messages/models.py]:
```python
group_names = filter(len, map(
    strip, ...
))
```

File: invenio/modules/messages/models.py [hunk 3]
```diff
@@ -85,108 +94,121 @@ def validate_sent_to_user_nicks(self, key, value):
...
+                            self.recipients.append(u)
 
     @property
     def user_nicks(self):
+        """User nicks."""
         if not self._sent_to_user_nicks:
             return []
-        return filter(len, map(strip,
-            self._sent_to_user_nicks.split(cfg['CFG_WEBMESSAGE_SEPARATOR'])))
+        return filter(
```
Comment (jirikuncar) [inline invenio/modules/messages/models.py]:
```python
return filter(len, map(
    strip, ...
))
```

File: invenio/modules/messages/models.py [hunk 4]
```diff
@@ -85,108 +94,121 @@ def validate_sent_to_user_nicks(self, key, value):
...
+    is_reminder = m.received_date is not None and \
+        m.received_date > datetime.now()
 
     alert = send_email
     if is_reminder:
-        alert = lambda *args, **kwargs: scheduled_send_email(*args,
-                    other_bibtasklet_arguments=[
-                        m.received_date.strftime(datetext_format)],
-                    **kwargs)
+        alert = lambda *args, **kwargs: scheduled_send_email(
```
Comment (jirikuncar) [inline invenio/modules/messages/models.py]: Maybe `def alert(*args, **kwargs):` would be prefered ...

File: invenio/modules/accounts/models.py [hunk 5]
```diff
@@ -252,8 +252,8 @@ def __str__(self):
         nullable=False, server_default='INTERNAL')
 
     # FIXME Unique(login_method(70), name)
-    __table_args__ = (db.Index('login_method_name', login_method, name,
-                               mysql_length=[60, None]),
+    __table_args__ = (db.Index('login_method_name', 'login_method', 'name',
+                               mysql_length={'login_method': 60, 'name': 255}),
```
Comment (jirikuncar) [inline invenio/modules/accounts/models.py]: :+1:

File: invenio/modules/pages/models.py [hunk 6]
```diff
@@ -37,11 +35,13 @@ class Page(db.Model):
...
+        db.Text().with_variant(db.Text(length=2**32-2), 'mysql'),
+        nullable=True)
     # Default is pages/templates/default.html
     template_name = db.Column(db.String(70), nullable=True)
     created = db.Column(db.DateTime(), nullable=False, default=datetime.now)
     last_modified = db.Column(db.DateTime(), nullable=False,
                               default=datetime.now, onupdate=datetime.now)
 
-__all__ = ['Page']
+__all__ = ('Page')
```
Comment (jirikuncar) [inline invenio/modules/pages/models.py]: `__all__ = ('Page', )` (missing comma if you want to define a tuple

File: invenio/modules/records/models.py [hunk 7]
```diff
@@ -202,13 +208,17 @@ def bibxxx(cls):
 
 for idx in range(100):
     Bibxxx = "Bib{0:02d}x".format(idx)
-    globals()[Bibxxx] = type(Bibxxx, (db.Model, BibxxxMixin), {})
+    globals_Bibxxx = type(Bibxxx, (db.Model, BibxxxMixin), {})
```
Comment (jirikuncar) [inline invenio/modules/records/models.py]: why did you split it into two lines? maybe can you just break the line inside brackets?

File: invenio/modules/records/models.py [hunk 8]
```diff
@@ -202,13 +208,17 @@ def bibxxx(cls):
...
+    globals()[Bibxxx] = globals_Bibxxx
     BibrecBibxxx = "BibrecBib{0:02d}x".format(idx)
     globals()[BibrecBibxxx] = type(BibrecBibxxx,
                                    (db.Model, BibrecBibxxxMixin), {})
 
     models += [Bibxxx, BibrecBibxxx]
 
+    Index('ix_{0:02d}x_value'.format(idx),
+          globals_Bibxxx.value, mysql_length=35)
```
Comment (jirikuncar) [inline invenio/modules/records/models.py]: :+1:

**Comment by hachreak** [comment]
ping @jirikuncar all green XD

**Pull request merged**